cmd/kubectl-gadget/traceloop.go

```diff
@@ -213,183 +135,206 @@ func runTraceloopStop(cmd *cobra.Command, args []string) error {
-	podname := args[1]
-	idx := args[2]
+		os.Exit(0)
 
-	client, err := k8sutil.NewClientsetFromConfigFlags(utils.KubernetesConfigFlags)
-	if err != nil {
-		return commonutils.WrapInErrSetupK8sClient(err)
+		return ""
 	}
 
-	pod, err := client.CoreV1().Pods(namespace).Get(context.TODO(), podname, metav1.GetOptions{})
-	if err != nil {
-		return fmt.Errorf("failed to get pod %s: %w", podname, err)
-	}
+	traces := traceList.Items
+	for _, trace := range traces {
+		var infos []traceloopTypes.TraceloopInfo
 
-	if pod.Spec.NodeName == "" {
-		return fmt.Errorf("pod %s not scheduled yet", podname)
-	}
+		err = json.Unmarshal([]byte(trace.Status.Output), &infos)
+		if err != nil {
+			return err
+		}
 
-	fmt.Printf("%s", utils.ExecPodSimple(client, pod.Spec.NodeName,
-		fmt.Sprintf(`curl --silent --unix-socket /run/traceloop.socket 'http://localhost/dump-pod?namespace=%s&podname=%s&idx=%s' ; echo`,
-			namespace, podname, idx)))
+		containerID := ""
+		for _, info := range infos {
+			// The CLI can give a shorter ID (e.g. 12 characters long), so we need to
+			// test if the long container ID starts with the short one to use the long
+			// one in the rest of the code.
+			if strings.HasPrefix(info.ContainerID, id) {
+				containerID = info.ContainerID
 
-	return nil
-}
+				break
+			}
+		}
 
-func runTraceloopClose(cmd *cobra.Command, args []string) error {
-	if len(args) != 1 {
-		return commonutils.WrapInErrMissingArgs("<trace-name>")
-	}
+		// The container of interest does not depend on the current trace.
+		if len(containerID) == 0 {
+			continue
+		}
 
-	client, err := k8sutil.NewClientsetFromConfigFlags(utils.KubernetesConfigFlags)
-	if err != nil {
-		return commonutils.WrapInErrSetupK8sClient(err)
-	}
+		var err error
+
+		// We want to create only stream trace which is on the same node than
+		// corresponding trace.
+		params.Node = trace.Spec.Node
+
+		traceID, err = utils.CreateTrace(&utils.TraceConfig{
+			GadgetName:       "traceloop",
+			Operation:        gadgetv1alpha1.OperationCollect,
+			TraceOutputMode:  gadgetv1alpha1.TraceOutputModeStream,
+			TraceOutputState: gadgetv1alpha1.TraceStateCompleted,
+			CommonFlags:      &params,
+			Parameters: map[string]string{
+				// We will not create a traceloop/gadget.go:Trace for this specific
+				// trace CRD.
+				// In place, we will use the traceloop/gadget.go:Trace which was
+				// created when called traceloop/gadget.go:Start().
+				// To do so, we use this name to get the traceloop/gadget.go:Trace
+				// from the map.
+				// Nonetheless when Start()'ed, the used name was the namespaced one:
+				// https://github.com/inspektor-gadget/inspektor-gadget/blob/9532d507bbd741f6202e1945db20cb6d1471e0ac/pkg/controllers/trace_controller.go#L253
+				// So, we need to use the namespace here too.
+				"name":        fmt.Sprintf("%s/%s", trace.Namespace, trace.Name),
+				"containerID": containerID,
+			},
+			AdditionalLabels: map[string]string{
+				"type": "collecting",
```

Comment:
I think those strings ("type", "collecting", "global") should be constants in a package shared between the CLI and the gadget. It could be in a single file `pkg/gadget-collection/gadgets/traceloop/config/annotations.go`

Some other gadgets also have configuration that are hardcoded, like minlatency in fsslower:
```
$ git grep -n '"minlatency"'
```

This can be in a future PR.